tests/Unit/LogRequestTest.php

```diff
@@ -157,4 +161,22 @@ public function it_tests()
         $this->assertEquals('[ MASKED ]', $loggedHeaders['authorization'][0]);
         $this->assertEquals('Not Secret', $loggedHeaders['something-else'][0]);
     }
+
+    /** @test */
+    public function it_saves_request_to_db_immediately_when_received()
+    {
+        // Arrange
+        $req = Request::create("something");
+        $service = new RequestLogOptionsService();
+        $requestLog = new LogRequest($service);
+
+        // Act -> mock closure given to ensure terminate function is not hit
+        $requestLog->handle($req, function ($r) {});
+
+        // Assert
+        $this->assertDatabaseCount('request_logs', 1);
+        $reqLog = RequestLog::query()->firstOrFail();
+        $this->assertEquals(0, $reqLog->status);
+        $this->assertEquals("http://localhost/something", $reqLog->url);
```

Comment:
Evt. brug `$this->assertDatabaseHas()` :)

Reply:
Det ser dejligt mere kompakt ud :)

Reply:
Jep, og så slipper man også for $reqLog-variablen :)